tests/library/test_metrics.py

```diff
@@ -478,3 +670,163 @@ def _test_confidence_interval(self, metric, expected_ci_low, expected_ci_high):
+        )
+
+        self._test_grouped_instance_confidence_interval(
+            metric=FixedGroupMeanBaselineStringContainment(),
+            expected_ci_low=0.25,
+            expected_ci_high=1.0,
+        )
+
+        self._test_grouped_instance_confidence_interval(
+            metric=FixedGroupMeanParaphraseStringContainment(),
+            expected_ci_low=0.5,
+            expected_ci_high=0.6666666666666666,
+        )
+
+        self._test_grouped_instance_confidence_interval(
+            metric=FixedGroupNormCohensHParaphraseAccuracy(),
+            expected_ci_low=-1.0,
+            expected_ci_high=0.33333333333333337,
+        )
+
+        # note, this metric has an issue where the ci_high on PCs on Travis slightly diverges from the local results
+        # hence this test may fail on a PC
+        self._test_grouped_instance_confidence_interval(
+            metric=FixedGroupNormCohensHParaphraseStringContainment(),
+            expected_ci_low=-0.49999999999999994,
+            expected_ci_high=-0.39182655203060723,
+        )
+
+        self._test_grouped_instance_confidence_interval(
+            metric=FixedGroupPDRParaphraseAccuracy(),
+            expected_ci_low=0.6666666666666666,
+            expected_ci_high=1.0,
+        )
+
+        self._test_grouped_instance_confidence_interval(
+            metric=FixedGroupPDRParaphraseStringContainment(),
+            expected_ci_low=0.3333333333333333,
+            expected_ci_high=0.5,
+        )
+
+        self._test_grouped_instance_confidence_interval(
+            metric=FixedGroupNormHedgesGParaphraseAccuracy(),
+            expected_ci_low=-1.0,
+            expected_ci_high=0.01892225367237965,
+        )
+
+        self._test_grouped_instance_confidence_interval(
+            metric=FixedGroupNormHedgesGParaphraseStringContainment(),
+            expected_ci_low=-0.09757387538180902,
+            expected_ci_high=-0.046656947481584346,
+        )
+
+        # pass global dict because there are additional fields other than the main score
+        self._test_grouped_instance_confidence_interval(
+            metric=GroupMeanTokenOverlap(),
+            expected_global_result={
+                "group_mean_recall": 0.525,
+                "group_mean_f1": 0.5083333333333333,
+                "score": 0.5083333333333333,
+                "score_name": "group_mean_f1",
+                "group_mean_precision": 0.5,
+                "group_mean_recall_ci_low": 0.25,
+                "group_mean_recall_ci_high": 0.7083333333333334,
+                "group_mean_f1_ci_low": 0.22302503471948287,
+                "group_mean_f1_ci_high": 0.6805555555555555,
+                "score_ci_low": 0.22302503471948287,
+                "score_ci_high": 0.6805555555555555,
+                "group_mean_precision_ci_low": 0.20949399775845196,
+                "group_mean_precision_ci_high": 0.6666666666666666,
+            },
+        )
+
+    def _test_grouped_instance_confidence_interval(
```

Comment:
Same comment as above:

if these are the same tests as in [grouped_instance_metrics.py](https://github.com/IBM/unitxt/pull/452/files?file-filters%5B%5D=.py&show-viewed-files=true#diff-a3f7cbec6d34ae3240ec3ad5e6198dc6f2ccc1ab9edb5fb47bcda4595e246285), then its not needed to do this twice... the grouped_instance_metrics.py runs as part of the tests, so its ok to assume that repeating what it does is not needed..

Reply:
Elron said this is ok